feature/posts/src/commonMain/kotlin/com/tunjid/heron/posts/PostsScreen.kt

```diff
@@ -16,26 +16,323 @@
+                    },
+                state = pullToRefreshState,
+                isRefreshing = state.isRefreshing,
+                onDismissRequest = {
+                    // Handle refresh dismissal if needed
+                },
+            )
+        },
     ) {
-        Text("Posts Screen")
+        LookaheadScope {
+            LazyVerticalStaggeredGrid(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .onSizeChanged {
+                        val itemWidth = with(density) {
+                            presentation.cardSize.toPx()
+                        }
+                        val numColumns = floor(it.width / itemWidth).roundToInt()
+                        if (numColumns > 0) actions(Action.Tile(TilingState.Action.GridSize(numColumns = numColumns)))
+                    },
+                state = gridState,
+                columns = StaggeredGridCells.Adaptive(presentation.cardSize),
+                verticalItemSpacing = presentation.lazyGridVerticalItemSpacing,
+                contentPadding = UiTokens.bottomNavAndInsetPaddingValues(
+                    top = UiTokens.statusBarHeight + UiTokens.toolbarHeight,
+                ),
+                horizontalArrangement = Arrangement.spacedBy(
+                    presentation.lazyGridHorizontalItemSpacing,
+                ),
+                userScrollEnabled = !paneScaffoldState.isTransitionActive,
+            ) {
+                items(
+                    items = items.distinctBy(TimelineItem::id),
+                    key = TimelineItem::id,
+                    itemContent = { item ->
+                        TimelineItem(
+                            modifier = Modifier
+                                .fillMaxWidth()
+                                .animateItem()
+                                .threadedVideoPosition(
+                                    state = videoStates.getOrCreateStateFor(item),
+                                ),
+                            paneMovableElementSharedTransitionScope = paneScaffoldState,
+                            presentationLookaheadScope = this@LookaheadScope,
+                            now = now,
+                            item = item,
+                            sharedElementPrefix = ShareElementPrefix,
+                            presentation = presentation,
+                            postActions = remember {
+                                postActions(
+                                    onLinkTargetClicked = { _, linkTarget ->
+                                        if (linkTarget is LinkTarget.Navigable) actions(
+                                            Action.Navigate.To(
+                                                pathDestination(
+                                                    path = linkTarget.path,
+                                                    referringRouteOption = NavigationAction.ReferringRouteOption.ParentOrCurrent,
```

Comment:
All usages of `NavigationAction.ReferringRouteOption.ParentOrCurrent` should be `NavigationAction.ReferringRouteOption.Current`